### PR TITLE
render within shadow DOM test added

### PR DESCRIPTION
**Pull request by gabrielo91**
# Description
- Adde new test for nested shadow DOM support

## Nested shadow DOM support PR

- https://github.com/krakenjs/belter/pull/54

### CODE REVIEW

**Comment by bluepnume** [comment]
Looks great! One thing I don't understand -- how is the test passing before https://github.com/krakenjs/belter/pull/54 is merged?

**Comment by gabrielo91** [comment]
> Looks great! One thing I don't understand -- how is the test passing before [krakenjs/belter#54](https://github.com/krakenjs/belter/pull/54) is merged?

Still wondering the same, today later we can discuss the approach 😅

File: test/tests/happy.jsx [hunk 1]
```diff
@@ -440,6 +440,34 @@ describe('zoid happy cases', () => {
...
+                });
+            };
+
+            onWindowOpen().then(expect('onWindowOpen', ({ win }) => {
+                if (getParent(win) !== window) {
+                    throw new Error(`Expected window parent to be current window`);
+                }
+            }));
+
+            const { container, destroy } = getContainer({ shadow: true, nested: true });
```
Comment (gregjopa) [inline test/tests/happy.jsx:460]: @gabrielo91 and I were pairing on this and noticed that `slot: true` is missing. Adding it here helps the new nested use case to get hit. 

```suggestion
            const { container, destroy } = getContainer({ shadow: true, nested: true, slot: true });
```

File: test/common.js [hunk 2]
```diff
@@ -211,6 +219,32 @@ export function getContainer({ parent, shadow = false, slots = false } : {| pare
         };
     }
 
+    if (nested) {
+        const innerWrapper = document.createElement('inner-element-wrapper');
+        innerWrapper.setAttribute('id', 'inner-element-wrapper-id');
+
+        const customElementShadowRoot = customElement.shadowRoot;
```
Comment (gregjopa) [inline test/common.js:226]: There is still some logic here that isn't working as expected. @gabrielo91 will continue to debug.

**Comment by bluepnume** [comment]
This is the first time I've ever commented on a PR to say "the tests are passing! please fix the PR so the tests break!" :D

File: test/common.js [hunk 3]
```diff
@@ -201,9 +202,43 @@ export function getContainer({ parent, shadow = false, slots = false } : {| pare
...
+
     const customElement = document.createElement(customElementName);
     parentContainer.appendChild(customElement);
 
+    if (nested) {
+        const innerWrapper = document.createElement(innerWrapperName);
+        innerWrapper.setAttribute('id', 'inner-element-wrapper-id');
+
+        const customElementShadowRoot = customElement.shadowRoot;
+        const innerWrapperShadowRoot = innerWrapper.shadowRoot;
```
Comment (gregjopa) [inline test/common.js:220]: The `innerWrapper` isn't in the DOM so we can't get at it's `shadowRoot` yet. Please try this instead:

```js
if (nested) {
        const innerWrapper = document.createElement(innerWrapperName);
        innerWrapper.setAttribute('id', 'inner-element-wrapper-id');

        const customElementShadowRoot = customElement.shadowRoot;

        if (customElementShadowRoot) {
            customElementShadowRoot.appendChild(innerWrapper);
        }

        const innerWrapperShadowRoot = innerWrapper.shadowRoot;
        const innerWrapperContainer = document.createElement('div');
        innerWrapperContainer.setAttribute('id', 'super-id');

        if (innerWrapperShadowRoot) {
            innerWrapperShadowRoot.appendChild(innerWrapperContainer);
        }

        return {
            container: innerWrapperContainer,
            destroy:   () => {
                parentContainer.removeChild(customElement);
                parentContainer.removeChild(innerWrapper);
            }
        };
    }
```

Comment (gregjopa) [reply test/common.js:220]: After this code change we should expect to get the following error message `Host element is also in shadow dom`:

expected-failure-message

**Review by gregjopa — APPROVED** [review]
Looks great! We are getting the expected error now. Upgrading to the latest belter should fix this test failure.

**Review by bluepnume — APPROVED** [review]
LGTM! Thanks Gabriel. This should give a lot of confidence for any component using zoid with shadow dom.

**Pull request merged**